### PR TITLE
Fix flaky subselect gp2

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
The PR contains 2 commit:
```
commit b8327b0664e95d4485cf098e6d58c621fa870841 (HEAD -> fix-flaky-subselect_gp2, bc/fix-flaky-subselect_gp2)
Author: Bhuvnesh Chaudhary <bhuvnesh2703@gmail.com>
Date:   Fri Oct 11 14:44:11 2019 -0700

    Revert "Move test partial_table to another test group to avoid deadlock"

    This reverts commit d9606d1808645c8776fe34335118c6f54e3a88fa.

    Refer to the commit: Use ANALYZE instead of VACUUM FULL for
    subselect_gp2 test which removes VACUUM FULL which caused the deadlock.

commit 1fcc2527f1b4387642cea04b178401041075a884
Author: Bhuvnesh Chaudhary <bhuvnesh2703@gmail.com>
Date:   Fri Oct 11 14:38:29 2019 -0700

    Use ANALYZE instead of VACUUM FULL for subselect_gp2 test

    There seems to be a random failure due to the deadlock which is caused
    by conflicting locks taken by VACUUM FULL pg_authid and delete / update
    operation on gp_distribution_policy. Instead of moving the test to a
    separate group, use ANALYZE instead of VACUUM to ensure that the
    estimates are mostly accurate.  This hopefully should keep the
    consistency and remove the overhead of locking by VACUUM FULL.

    There was a similar issue which was fixed via commit `d9606d180` which
    describes the deadlock as below:

    `update gp_distribution_policy`'s lock-acquire:
      1. at parsing stage, lock `gp_distribution_policy` in Exclusive Mode
        2. later when it needs to check authentication, lock `pg_authid` in
             AccessShare Mode

    `VACUUM FULL pg_authid`'s lock-acquire:
      1. lock pg_authid in Access Exclusive Mode
        2. later when rebuilding heap, it might delete some dependencies,
             this will do GpPolicyRemove, which locks `gp_distribution_policy`
                  in RowExclusive Mode

    So there is a potential local deadlock.

    Co-Authored-By: Ashwin Agrawal <aagrawal@pivotal.io>
```

### CODE REVIEW

File: src/test/regress/expected/subselect_gp2.out [hunk 1]
```diff
@@ -45,12 +45,12 @@ and usename='xxx' and datname='xxx';
...
-         ->  Function Scan on pg_stat_get_activity s
-               Filter: (query = $0)
+         Hash Cond: (d.oid = s.datid)
+         ->  Seq Scan on pg_database d
+               Filter: (datname = 'xxx'::name)
          ->  Hash
-               ->  Seq Scan on pg_database d
-                     Filter: (datname = 'xxx'::name)
+               ->  Function Scan on pg_stat_get_activity s
+                     Filter: (query = $0)
```
Comment (pivotal-ning-yu) [inline src/test/regress/expected/subselect_gp2.out]: Do we know why the join order is changed in the plan?  Will the order always be the same?

**Comment by paul-guo-** [comment]
@bhuvnesh2703 In another PR https://github.com/greenplum-db/gpdb/pull/8729/files I removed the requirement of either `vacuum full` or `analyze`. So any reason that this PR changes like this? Though it seems that with either PR the partial_table could be put back into the parallel group again?

**Comment by bchaudhary** [comment]
Perfect, we don’t need it.

**Comment by bhuvnesh2703** [comment]
Yes we could bring back the partial table one..

**Comment by paul-guo-** [comment]
OK. I revert d9606d1808645c8776fe34335118c6f54e3a88fa along with my PR.

**Pull request closed without merging**